### ISSUE TITLE
How to install Nix in private subnet?

### ISSUE THREAD

**Issue by vitaliy-zinchenko** [NixOS/nix#2787]
I need to install the nix-package manager to the EC2 instance in private subnet. 

The VPC endpoint policies allow to talk to S3. That is why I put (for now in my test bucket) the nix release package in S3 bucket. But the install script tries to download the default nix channel from the internet (https://nixos.org/channels/nixpkgs-unstable). And I didn't find the way to provide the custom nix channel in the install script (without modifying the install script).

How can we install nix package manager in private subnet?

**Comment by matthewbauer** [comment]
I don't think this is currently supported. I had an attempt to accomplish this previously, but never had enough time to get it working:

https://github.com/NixOS/nix/pull/2135

Alternatively, you can do something like this:

- Download tarball from https://nixos.org/releases/nix/nix-2.2.2/nix-2.2.2-x86_64-linux.tar.bz2
- Extract tarball into nix-2.2.2 directory
- Run ```nix-2.2.2/install``` to install
- Run ```nix-channel --add ... nixpkgs```

**Comment by bjornfor** [comment]
> But the install script tries to download the default nix channel from the internet [...]

A low-hanging fruit is merging https://github.com/NixOS/nix/pull/2679

I also attempted adding a "--channel URL" option to the installer script, but that was rejected: https://github.com/NixOS/nix/pull/1284/commits/013e5a80eb48c0b06c71b9149391d9b532e205d6.

**Comment by tomberek** [comment]
Closing as stale. Re-open if needed.

**Comment by bjornfor** [comment]
Fixed by https://github.com/NixOS/nix/pull/2679.
